### PR TITLE
chore: update to effet 3.10

### PR DESCRIPTION
**Pull request by titouancreach**
Effect has updated to 3.10 merging @effect/schema into effect main package.

https://effect.website/blog/effect-3.10

So this PR, remove @effect/schema, change the imports and update effect.

btw it deprecates: https://github.com/react-hook-form/resolvers/pull/710

### CODE REVIEW

File: effect-ts/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "peerDependencies": {
     "@hookform/resolvers": "^2.0.0",
     "@effect/schema": "^0.66.14",
-    "effect": "^3.1.2",
+    "effect": "^3.10.1",
```
Comment (vinassefranche) [inline effect-ts/package.json:16]: can't we remove the @effect/schema peer dependency, now?

Comment (titouancreach) [reply effect-ts/package.json:16]: sure we can

**Comment by titouancreach** [comment]
pretty sure there is something to change here too: https://github.com/react-hook-form/resolvers/blob/37c3f781c936fa2eb25eace737eefc6275fd09a4/package.json

**Comment by charlee-dev** [comment]
also, you can try to bump it to 3.10.2 instead, as they had another update last night

**Comment by akoenig** [comment]
@titouancreach I contributed a PR back to your fork with the update of the `microbundle` task and an update to the most recent version of Effect.

**Comment by titouancreach** [comment]
@akoenig great !! can you make a PR on my fork 🙃 So we don't loose the commit history

**Comment by dudeofawesome** [comment]
I've created [a PR](https://github.com/titouancreach/resolvers/pull/2) to bring the last of my changes from #710 that didn't make it into this PR.

**Comment by titouancreach** [comment]
Thanks @dudeofawesome !

**Comment by titouancreach** [comment]
@akoenig I don't know if it helps, but this resolver only consist in two files, I simply copy pasted it to my code waiting for this PR to be merged, so my code don't stay behind

**Comment by dudeofawesome** [comment]
@akoenig I've also got a version based on [my old PR](https://github.com/react-hook-form/resolvers/pull/710) [published on NPM](https://www.npmjs.com/package/@dudeofawesome/hookform-resolvers), which I've been using to keep up with Effect.

**Review by bluebill1049 — APPROVED** [review]

**Comment by lauren-inato** [comment]
@bluebill1049 any idea when this will be released?

**Pull request merged**